.eslintrc

```diff
@@ -1,66 +1,78 @@
 {
```

Comment:
Getting `eslint` and `typescript` to work is always a challenge. I'm glad that you went with `eslint` instead of `tslint`. This has been my go to config, using a configs and plugins from [`Sindre Sorhus`](https://github.com/xojs/eslint-config-xo-typescript).  

**package.json**
```json
  "dependencies": {
    "@typescript-eslint/eslint-plugin": "^1.13.0",
    "@typescript-eslint/parser": "^1.13.0",
    "eslint": "^6.1.0",
    "eslint-config-xo-react": "^0.20.0",
    "eslint-config-xo-space": "^0.21.0",
    "eslint-config-xo-typescript": "^0.15.0",
    "eslint-formatter-pretty": "^2.1.1",
    "eslint-plugin-eslint-plugin": "^2.1.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-jest": "^22.14.0",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.14.3",
    "eslint-plugin-react-hooks": "^1.6.1",
    "eslint-plugin-simple-import-sort": "^4.0.0",
    "eslint-plugin-unicorn": "^10.0.0"
  },
```

**eslint.config.js**
```javascript
module.exports = {
    root: true,
    plugins: [
        'simple-import-sort',
        'import',
        'eslint-plugin',
        'jest',
        'react-hooks',
        'promise',
        'unicorn',
    ],
    env: {
        es6: true,
        node: true,
    },
    extends: [
        'xo-space/esnext',
        'xo-react/space',
        'xo-typescript',
        'plugin:promise/recommended',
        'plugin:unicorn/recommended'
    ],
    rules: {
        'capitalized-comments': 'off',
        'comma-dangle': [
            'error',
            {
                arrays: 'only-multiline',
                exports: 'never',
                functions: 'never',
                imports: 'never',
                objects: 'only-multiline',
            },
        ],
        'no-console': 'off',
        '@typescript-eslint/indent': 'off',
        '@typescript-eslint/explicit-member-accessibility': 'off',
        '@typescript-eslint/no-use-before-define': 'error',
        '@typescript-eslint/no-parameter-properties': 'off',
        //
        'sort-imports': 'off',
        'import/order': 'off',
        'simple-import-sort/sort': 'warn',
        //
        'import/first': 'error',
        'import/newline-after-import': 'warn',
        'import/no-duplicates': 'error',
        //
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
        'unicorn/prevent-abbreviations': 'off',
        'unicorn/filename-case': [
            'error',
            {
                cases: {
                    camelCase: true,
                    pascalCase: true
                }
            }
        ]
    },
    overrides: [
        {
            env: {
                'jest/globals': true,
            },
            files: [
                'src/**/*.test.ts',
                'tests/**/*.test.ts',
            ],
            rules: {
                'jest/no-alias-methods': 'error',
                'jest/no-disabled-tests': 'warn',
                'jest/no-focused-tests': 'error',
                'jest/no-identical-title': 'error',
                'jest/no-jasmine-globals': 'error',
                'jest/no-jest-import': 'error',
                'jest/no-test-callback': 'error',
                'jest/no-test-prefixes': 'error',
                'jest/no-test-return-statement': 'error',
                'jest/prefer-spy-on': 'error',
                'jest/prefer-to-have-length': 'warn',
                'jest/valid-expect': 'error',
            },
        },
        {
            files: [
                'src/**/*.test.ts',
                'tests/**/*.test.ts',
            ],
            rules: {
                'eslint-plugin/no-identical-tests': 'error',
            },
        },
        {
            files: ['*.config.js', 'scripts/*.js'],
            rules: {
                '@typescript-eslint/no-var-requires': 'off'
            }
        }
    ],
};
```

Reply:
I'm open about the listing. I took the setup from [another project](https://github.com/camacho/node-repository-template/blob/master/.eslintrc.json) I am working on .

I'm open to anything as long as it includes:
- TypeScript official recommendations
- Jest  plugin and recommendations
- Import plugin and rules
- Prettier rules

I'd like to abide by Airbnb standards, but that's personal preference.

Reply:
They are transients. Your choices are good for the project. Mine was just more of a reference and it has others useful rules for import sorting and working React.